### PR TITLE
docs: Update API docs with codegen examples

### PR DESCRIPTION
**Pull request by yuth**
Updated the GraphQL API docs to use statements generated
by the codegen. Removed the `amplify codegen add` command as its
no longer neeed to be added

### CODE REVIEW

File: js/api.md [hunk 1]
```diff
@@ -114,20 +114,12 @@ The following transformers are available to be used with AWS AppSync when defini
 
 ##### Type Generation using GraphQL Schemas
 
-When working with GraphQL data, it is useful to import your types from your schema into your code for type safety. You can easily do this with Amplify CLI's automated code generation feature. The CLI automatically downloads GraphQL Introspection Schemas from your defined GraphQL endpoint and generates TypeScript or Flow classes for you.
+When working with GraphQL data, it is useful to import your types from your schema into your code for type safety. You can easily do this with Amplify CLI's automated code generation feature. The CLI automatically downloads GraphQL Introspection Schemas from your defined GraphQL endpoint and generates TypeScript or Flow classes for you. Every time you push your GraphQL API, the cli will confirm if you want to generate the types and statements.
```
Comment (undefobj) [inline js/api.md]: CLI in caps last sentence

File: js/api.md [hunk 2]
```diff
@@ -256,102 +248,47 @@ AWS Amplify API category provides a GraphQL client for working with queries, mut
 
 #### Query Declarations
 
-You can declare GraphQL queries with standard GraphQL query syntax. To learn more about GraphQL queries, please visit [GraphQL Developer Documentation](http://graphql.org/learn/queries/). 
-
+Amplify cli codegen automatically generates all possible GraphQL statements (queries, mutations and subscriptions) and saves it in `src/graphql` folder.
```
Comment (undefobj) [inline js/api.md]: ...and for JavaScript applications saves it in `src/graphql` folder

File: js/api.md [hunk 3]
```diff
@@ -456,33 +381,112 @@ Also, you can use `subscription` and `onSubscriptionMsg` attributes to enable su
...
+    {({ data: { listTodos }, loading, error }) => {
+        if (error) return (<h3>Error</h3>);
+        if (loading || !listTodos) return (<h3>Loading...</h3>);
+        <ListView todos={listTodos ? listTodos.items : []} />
+    }
  </Connect>
 
 ```
 
 For mutations, a `mutation` function needs to be provided with `Connect` component. `mutation` returns a promise that resolves with the result of the GraphQL mutation.
```
Comment (mlabieniec) [inline js/api.md]: update to:

For mutations, a `mutation` function needs to be provided with the `Connect` component. A `mutation` returns a promise that resolves with the result of the GraphQL mutation.

File: js/api.md [hunk 4]
```diff
@@ -114,20 +114,12 @@ The following transformers are available to be used with AWS AppSync when defini
 
 ##### Type Generation using GraphQL Schemas
 
-When working with GraphQL data, it is useful to import your types from your schema into your code for type safety. You can easily do this with Amplify CLI's automated code generation feature. The CLI automatically downloads GraphQL Introspection Schemas from your defined GraphQL endpoint and generates TypeScript or Flow classes for you.
+When working with GraphQL data, it is useful to import your types from your schema into your code for type safety. You can easily do this with Amplify CLI's automated code generation feature. The CLI automatically downloads GraphQL Introspection Schemas from your defined GraphQL endpoint and generates TypeScript or Flow classes for you. Every time you push your GraphQL API, the CLI will confirm if you want to generate the types and statements.
```
Comment (mlabieniec) [inline js/api.md]: change to:

When working with GraphQL data it is useful to import types from your schema for type safety. You can do this with the Amplify CLI's automated code generation feature. The CLI automatically downloads GraphQL Introspection Schemas from the defined GraphQL endpoint and generates TypeScript or Flow classes for you. Every time you push your GraphQL API, the CLI will provide you the option to generate types and statements.

**Comment by undefobj** [comment]
Looks good to me - @elorzafe can you please confirm that we have the `loading` fix already deployed?

**Comment by elorzafe** [comment]
That is not merged yet, I was reviewing it with Manuel, we can merged and publish today

**Pull request merged**